Sources/Datadog/DatadogConfiguration.swift

```diff
@@ -454,10 +454,10 @@ extension Datadog {
             }
 
             /// Sets the custom mapper for `RUMViewEvent`. This can be used to modify RUM View events before they are send to Datadog.
-            /// - Parameter mapper: the closure taking `RUMViewEvent` as input and expecting `RUMViewEvent` or `nil` as output.
-            /// The implementation should obtain a mutable version of the `RUMViewEvent`, modify it and return. Returning `nil` will result
-            /// with dropping the RUM View event entirely, so it won't be send to Datadog.
-            public func setRUMViewEventMapper(_ mapper: @escaping (RUMViewEvent) -> RUMViewEvent?) -> Builder {
+            /// - Parameter mapper: the closure taking `RUMViewEvent` as input and expecting `RUMViewEvent` as output.
+            /// The implementation should obtain a mutable version of the `RUMViewEvent`, modify it and return it.
+            /// Note that the mapper intentionally prevents from returning a `nil` to drop the `RUMViewEvent` entirely, this ensures that all `RUMViewEvent` are sent to Datadog.
```

Comment:
if i were a user, that could be a bit confusing to me
i have `ViewPredicate` where I can keep or drop views and then i also have `viewEventMapper` which intentionally prevents me from dropping them.

i have a dilemma here:
1. if we mention `ViewPredicate` here and how it's different than `viewMapper`, etc. this may be even more confusing in case the user never saw `ViewPredicate` API before
2. if we don't mention it, this may be still confusing if the user is already aware of `ViewPredicate` API

what do you think about that?

Reply:
Those are good points and that can look confusing.

My understanding is that the `ViewPredicate` gives an opportunity to drop views already, and downstream the `RUMViewEventMapper` will be fed with events based on views that have been already filtered.
Before that PR, we give the opportunity (again) to drop view related data at the `RUMViewEventMapper`, I think it can create a confusing pipeline (the other way around regarding your concerns).

My point is that `ViewPredicate` let's the user code filter out views, there will be no events for already filtered out views and `RUMViewEventMapper` will only receive events for views that have been chosen already.

Does that make sense?  

That being said, maybe it would help to mention the `UIKitRUMViewsPredicate` here for context.

Reply:
I'd vote for mentioning the `UIKitRUMViewsPredicate` in there.

Using this scrubbing API to ignore particular `UIViewControllers` from being tracked is a misuse of the SDK APIs. It means the user doesn't understand the SDK auto instrumentation & scrubbing APIs well and we should guide him towards the success.

That said, this change to "view events cannot be dropped" makes the contract between auto instrumentation & scrubbing APIs even more solid and puts the user on pathway to success.